### PR TITLE
Ensure buildversion.props is evaluated before dotnet restore

### PR DESCRIPTION
**Pull request by chcosta**

### CODE REVIEW

File: build.proj [hunk 1]
```diff
@@ -56,8 +56,7 @@
   </Target>
 
   <Target Name="BatchRestorePackages" Condition="'$(RestoreDuringBuild)'=='true'">
-    <Message Importance="High" Text="Restoring all packages..." />
-    <Exec Condition="'@(SdkRestoreProjects)' != ''" Command="$(DotnetRestoreCommand) &quot;%(SdkRestoreProjects.FullPath)&quot; %(SdkRestoreProjects.ExtraRestoreArgs)" StandardOutputImportance="Low" />
+    <MSBuild Projects="$(MSBuildThisFileDirectory)restore.proj" />
```
Comment (eerhardt) [inline build.proj]: I'd add a comment to why we are doing this here.

**Comment by Petermarcu** [comment]
Can we get a PR open for release/2.0.0 if you guys think this is the right fix?

**Comment by chcosta** [comment]
I'll prepare a PR now

**Comment by gkhanna79** [comment]
Given that 2.0.0 fix is merged, should this be merged as well?

**Comment by eerhardt** [comment]
Merging.  I'll update my parallel PR to merge with this.  @chcosta and I discussed offline where we want to go in this area.

**Pull request merged**